### ISSUE TITLE
[git] re-sign previous commits to fix push declined

### ISSUE THREAD

**Issue by nguyenvulong** [nguyenvulong/QA#92]
### error message
``` 
! [remote rejected] main -> main (push declined due to repository rule violations)
error: failed to push some refs to 'github.com:nguyenvulong/sound-localization.git'
```

### solution
```
git rebase -i --exec 'git commit --amend --no-edit -S' origin/main
git push
```

The command `git rebase -i --exec 'git commit --amend --no-edit -S' origin/main` is performing an interactive rebase starting from the shared branch `origin/main` to the current branch you are on. Here's a detailed explanation of each component:

- `git rebase -i`: This part initiates an interactive rebase. Interactive rebases allow you to edit the commit history by specifying actions for each commit, such as reordering, squashing, or modifying commits.

- `--exec 'git commit --amend --no-edit -S'`: This option specifies a command to be automatically executed for each commit during the rebase process. The specified command here is adjusting each commit by amending it:
  - `git commit --amend --no-edit`: The `--amend` option lets you change the last commit. Since it’s used with `--no-edit`, it doesn't open the editor to modify the commit message, keeping it the same.
  - `-S`: This option signs the commit, assuming you have configured Git with your GPG key or SSH key.

In summary, this command goes through each commit in your current branch that isn't in `origin/main`, and it amends each commit to be GPG-signed while keeping the commit message unchanged. The interactive part of the rebase (`-i`) allows you to tweak other aspects of the commits or resolve any conflicts that may arise during the rebase.

### credit: [warp.dev](https://app.warp.dev/referral/REQYP5)
